File: SchoolManage/SMS/SMS/SMS/ClientApp/src/app/master/student/student.ts
import { Address } from "../../shared/address/address";

export class Student {
    aadharNumber? : string;
    academicPrecentage? : string;
    addresses ?: Address[] ;
    admissionNumber ?: string;
    amissionDate?: string;
    bloodGroup?: string;
    class?: string;
    dob?: string;
    emailId?: string;
    emisNumber?: string;
    firstLanguage?: string;
    firstName?: string;
    gender?: string;
    lastName?: string;
    middleName?: string;
    mobile?: string;
    salutation?: string;
    nationalityId?: string;
    passingOutSchool?: string;
    reasonForLeaving?: string;
    religionId?: string;
    rollNo?: string;
    schoolBrand?: string;
    schoolName?: string;
    secondLanguage?: string;
    section?: string;
    yearofattendence?: string;
    fatherDetails?: Dependents;
    motherDetails?: Dependents;
    localGuardian?: Dependents;
    legalGuardian?: Dependents;
}

export class Dependents {
    aadharNumber?: string;
    annualIncome?: string;
    bvEmployee?: boolean;
    company?: string;
    designation?: string;
    email?: string;
    firstName?: string;
    lastName?: string;
    middleName?: string;
    mobileNumber?: string;
    occupation?: string;
    salutation?: string;
}

File: SchoolManage/SMS/SMS/SMS/Models/Student/Student.cs
using SMS.Models.Transport;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Reflection.Metadata;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

#nullable disable

namespace SMS.Models
{
  public class Student
  {

    [JsonIgnore]
    [IgnoreDataMember]
    [DatabaseGenerated(DatabaseGeneratedOption.None)]
    public int StudentId { get; set; }



    //[JsonIgnore]
    //[IgnoreDataMember]
    //public StudentUserCred StudentUserCred { get; set; }

    public int? Salutation { get; set; }
    public string FirstName { get; set; }
    public string MiddleName { get; set; }
    public string LastName { get; set; }
    public DateTime Dob { get; set; }
    public int? BloodGroup { get; set; }
    public int? NationalityId { get; set; }
    public int? ReligionId { get; set; }
    public int? Gender { get; set; }
    public string EmailId { get; set; }
    public string AadharNumber { get; set; }

    [JsonIgnore]
    [IgnoreDataMember]
    public int StudentAddressId { get; set; }

    //[JsonIgnore]
    //[IgnoreDataMember]
    //public StudentAddress[] Addresses { get; set; }

    public virtual ICollection<StudentAddress> Addresses { get; set; }

    public long Mobile { get; set; }
    public int AdmissionNumber { get; set; }
    public DateTime AdmissionDate { get; set; }
    public string Class { get; set; }
    public string Section { get; set; }
    public string RollNo { get; set; }
    public int? FirstLanguage { get; set; }
    public int? SecondLanguage { get; set; }
    public string EmisNumber { get; set; }
    public string schoolName { get; set; }
    public int? schoolBrand { get; set; }
    public int? passingOutSchool { get; set; }
    public int yearofattendence { get; set; }
    public string AcademicPrecentage { get; set; }
    public string ReasonForLeaving { get; set; }

    [JsonIgnore]
    [IgnoreDataMember]
    public int DependentsId { get; set; }

    public virtual ICollection<Dependents> Dependentsdetails { get; set; }

    [JsonIgnore]
    [IgnoreDataMember]
    public int BusTripid { get; set; }

    //[JsonIgnore]
    //[IgnoreDataMember]
    //public virtual BusTrip StudentTrip { get; set; }

  }
}
